apiml-security-common/src/main/java/org/zowe/apiml/security/common/filter/CategorizeCertsFilter.java

```diff
@@ -142,11 +142,26 @@ protected void doFilterInternal(@NonNull HttpServletRequest request, @NonNull Ht
         filterChain.doFilter(mutate(request), response);
     }
 
-    private X509Certificate[] selectCerts(X509Certificate[] certs, Predicate<X509Certificate> test) {
-        if (test.test(certs[0])) {
-            return certs;
+    /**
+     * Selects certificates from an array based on a predicate.
+     * Only certificates that match the predicate will be included in the result array.
+     * IMPORTANT: This method replicates the original filter's logic.
+     * <p>
+     * Null certificates are automatically excluded.
+     *
+     * @param certs The array of X.509 certificates to be filtered.
+     * @param test  The predicate to test each certificate against.
+     * @return A new array containing only the certificates that match the predicate.
+     */
+    public static X509Certificate[] selectCerts(X509Certificate[] certs, Predicate<X509Certificate> test) {
+        if (certs == null || certs.length == 0) {
```

Comment:
```suggestion
        if (ArrayUtils.isEmpty(certs)) {
```

Actually, it is just solving the issue of stream below, see:

Optional.ofNullable(certs).map(Arrays::stream).orElse(Stream.empty())...

Reply:
thanks for the suggestion, I will use the `Optional.ofNullable(certs).map`